py/desitarget/mock/build.py

```diff
@@ -394,7 +394,7 @@ def density_fluctuations(data, log, nside, nside_chunk, seed=None):
     areaperpixel = hp.nside2pixarea(nside, degrees=True)
     areaperchunk = hp.nside2pixarea(nside_chunk, degrees=True)
 
-    nchunk = 4**np.int(np.log2(nside_chunk) - np.log2(nside))
+    nchunk = 4**int(np.log2(nside_chunk) - np.log2(nside))
```

Comment:
Are you absolutely confident that this is never called with an array `nside_chunk`, in which case the `int(...)` cast would fail?  I think it would be safer to phrase this as
```
nchunk = 4**(np.log2(nside_chunk) - np.log2(nside)).astype(int)
```
which works whether `nside_chunk` is a vector or scalar, which could buy us some future robustness even if `nside_chunk` is strictly a scalar right now.

Reply:
That case _currently_ fails, I think, for an array (see my example code under TMI): 

```
nside_chunk1, nside1 = np.array([32, 32, 64]), np.array([16, 8, 16])
nside_chunk2, nside2 = np.array([32]), np.array([16])
nside_chunk3, nside3 = 32, 16
for nside_chunk, nside in zip([nside_chunk1, nside_chunk2, nside_chunk3], [nside1, nside2, nside3]):
    try:
        nchunk = 4**np.int(np.log2(nside_chunk) - np.log2(nside))
        print(nside_chunk, nchunk)
    except TypeError:
        print("Casting {} and {} wasn't possible anyway".format(nside_chunk, nside))

<ipython-input-4-9ca0a781b651>:6: DeprecationWarning: `np.int` is a deprecated alias for the builtin `int`. To silence this warning, use `int` by itself. Doing this will not modify any behavior and is safe. When replacing `np.int`, you may wish to use e.g. `np.int64` or `np.int32` to specify the precision. If you wish to review your current use, check the release note link for additional information.
Deprecated in NumPy 1.20; for more details and guidance: https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
  nchunk = 4**np.int(np.log2(nside_chunk) - np.log2(nside))
Casting [32 32 64] and [16  8 16] wasn't possible anyway
[32] 4
32 4
```
If the intention is for that case to succeed, it should not have been constructed using `np.int` in the first place.

Also, from [the docstring](https://github.com/desihub/desitarget/blob/692bd313e66720232f46ea47b6ac7104631380ab/py/desitarget/mock/build.py#L18-L35):

```
    nside : :class:`int`                                                                                                                   
        Healpix resolution.                                                                                                                
    nside_chunk : :class:`int`                                                                                                             
        Healpix resolution for chunking the sample.                                                                                        
```
Hence my near-absolute confidence.

Reply:
Also, for further confidence:

[This line](https://github.com/desihub/desitarget/blob/692bd313e66720232f46ea47b6ac7104631380ab/py/desitarget/mock/build.py#L390) would fail if `nside` or `nside_chunk` were array-like.